### ISSUE TITLE
[Feature Request] Add TeX as an PDF

### ISSUE THREAD

**Issue by sambhavsaggi** [xournalpp/xournalpp#375]
Currently, when you insert a TeX equation, it defaults to an ugly, pixely mess. Currently, Mathtex only supports PNGs (i think) and we can replace this with a better engine.

**Comment by andreasb242** [comment]
I fixed some bugs on Mathtex, and I think the main issue is the resolution calculation, which is really bad.

in
>bool LatexController::runCommand()

If you insert an image, there is no old image, and therefore:
`
	if (this->texArea < 1000)
	{
		texres = "300";
	}
`
I'm not sure, but this might be the issue.

**Comment by Rmano** [comment]
If you want to use just LaTeX to generate a PDF, what I use is normally a standalone class construct, where you have to embed the equation string into this (in this case the string is the equation starting with `a=`):

```latex
\documentclass[border=5pt]{standalone}
\usepackage{amsmath}
\begin{document}
\(\displaystyle
      a=\int_0^\infty f(x)d\,x
\)
\end{document}
```
...and then simply run `pdflatex` on it, you'll have a cropped  pfd: 

![image](https://user-images.githubusercontent.com/6414907/49070798-cb69ee00-f22c-11e8-8f0d-4e2a2a71f483.png)

The problem here is deciding which packages you need/want for your math mode strings (LaTeX is more complex than MathJax...)

**Comment by andreasb242** [comment]
I know latex, but not so good.
PDF would probably be easier than svg, as we already support PDF drawing to the page.

May the template can be overwritten in the settings? So you enter your latex template with some placeholder?
E.g. also a placeholder for the Color code?

**Comment by andreasb242** [comment]
So the idea: Make the path for `pdflatex` configurable, check it on the first usage. If not there, notify the user. This is also working on Windows.

Then use the GLib functions to executable, so it will work on other Plattforms.
https://developer.gnome.org/glib/stable/glib-Spawning-Processes.html

mathtex.c can then be removed.
Would this be Ok for anybody? I don't implement it now...

**Comment by andreasb242** [comment]
Technical now the PDF is loaded, but to the document still the image is saved / rendered. Therefore loading / saving, Model and drawing needs to be adapted. Drawing should not be an issue, as it's more or less the same as we already do on PDF rendering.
Embedding the PDF will not be that easy, probably also as hex, or live re-render latex...
I have to check, I probably save it as base64 to the .xopp...

**Comment by morrolinux** [comment]
Would we still be able to copy/paste the generated latex as an image to another application after that? 
because that would be a very convenient feature. Right now the png resolution is quite high and you don't really get pixelated stuff unless you fill an A4 page with a single formula, so I wouldn't mind the actual png format. just my 2 cents

**Comment by andreasb242** [comment]
Copy Paste is never an issue, as this anyway is copied to clipboard in different formats, e.g. as image.

**Comment by Rmano** [comment]
@morrolinux  yes, probably the high-resolution PNG is ok most of the time, and when rotation for images is implemented, that will be ok for me too. 
@andreasb242 , if you need info for the templates (and colors, and whatever) just mention me, I'll do my best to help (and to keep pace with your super speed in coding... ;-) )

**Comment by andreasb242** [comment]
@Rmano  haha, thank you!
I know latex a little, I had to use it for my bachelor thesis... ;-)

But yes, I see you are a specialist.

I still plan to use PDF for rendering. Copy to clipboard is anyway rendered, else stroke could not be copied. So there is no issue for copy and paste.

But my plan is currently to do a release first. But It seems the MacBook where I created the release last time is died... And my second MacBook is reaaaaaallllllyyyy sssslllloooowwww.... ;-)
